include/deal.II/arborx/access_traits.h

```diff
@@ -0,0 +1,275 @@
+                              static_cast<float>(dim_points[i][1]),
+                              dim == 2 ? 0. :
+                                         static_cast<float>(dim_points[i][2]));
+        }
+    }
+
+    /**
+     * Number of points stored in the structure.
+     */
+    std::size_t
+    size() const
+    {
+      return points.size();
+    }
+
+    std::vector<dealii::Point<3, float>> points;
+  };
+
+
+
+  /**
+   *  This structure is used by ArborXWrappers::BVH to determine which bounding
+   *  boxes intersect the bounding boxes used to build the ArborXWrappers::BVH.
+   */
+  struct BoundingBoxIntersect
+  {
+    /**
+     * Constructor. @p bb is a list of bounding boxes which we are interested in
+     * knowing if they intersect ArborXWrappers::BVH bounding boxes.
+     */
+    template <int dim, typename Number>
+    BoundingBoxIntersect(
+      const std::vector<dealii::BoundingBox<dim, Number>> &bb)
+    {
+      const unsigned int size = bb.size();
+      bounding_boxes.reserve(size);
+      dealii::Point<3, float> min_corner_arborx(0., 0., 0.);
+      dealii::Point<3, float> max_corner_arborx(0., 0., 0.);
+      for (unsigned int i = 0; i < size; ++i)
+        {
+          auto boundary_points                  = bb[i].get_boundary_points();
+          dealii::Point<dim, Number> min_corner = boundary_points.first;
+          dealii::Point<dim, Number> max_corner = boundary_points.second;
+          for (int d = 0; d < dim; ++d)
+            {
+              min_corner_arborx[d] = static_cast<float>(min_corner[d]);
+              max_corner_arborx[d] = static_cast<float>(max_corner[d]);
+            }
+          bounding_boxes.emplace_back(
+            std::make_pair(min_corner_arborx, max_corner_arborx));
+        }
+    }
+
+    /**
+     * Number of bounding boxes stored in the structure.
+     */
+    std::size_t
+    size() const
+    {
+      return bounding_boxes.size();
+    }
+
+    std::vector<dealii::BoundingBox<3, float>> bounding_boxes;
+  };
+} // namespace ArborXWrappers
+
+DEAL_II_NAMESPACE_CLOSE
+
+/**
+ * This namespace contains the implementation of AccessTraits used by ArborX.
+ */
+namespace ArborX
+{
+  /**
+   * This struct allows ArborX to use std::vector<dealii::BoundingBox> as
+   * primitive.
+   */
+  template <int dim, typename Number>
+  struct AccessTraits<std::vector<dealii::BoundingBox<dim, Number>>,
+                      PrimitivesTag>
+  {
+    static_assert(dim != 1, "dim equal to one is not supported.");
+
+    /**
+     * Return the size of the vector @p v.
+     */
+    static std::size_t
+    size(const std::vector<dealii::BoundingBox<dim, Number>> &v)
+    {
+      return v.size();
+    }
+
+    /**
+     * Return an ArborX::Box from the dealii::BoundingBox v[i].
+     */
+    static Box
+    get(const std::vector<dealii::BoundingBox<dim, Number>> &v, std::size_t i)
+    {
+      auto                       boundary_points = v[i].get_boundary_points();
```

Comment:
```suggestion
      const auto                       boundary_points = v[i].get_boundary_points();
```
same for the next two variables